libs/wingc/src/type_check.rs

```diff
@@ -2076,10 +2118,43 @@ impl<'a> TypeChecker<'a> {
 			.iter()
 			.take(func_sig.parameters.len() - num_optionals);
 
-		// Verify passed positional arguments match the function's parameter types
-		for (arg_expr, arg_type, param) in izip!(arg_list.pos_args.iter(), arg_list_types.pos_args.iter(), params) {
-			self.validate_type(*arg_type, param.typeref, arg_expr);
-		}
+		if index_last_item == arg_list_types.pos_args.len() {
+			for (arg_expr, arg_type, param) in izip!(arg_list.pos_args.iter(), arg_list_types.pos_args.iter(), params) {
+				self.validate_type(*arg_type, param.typeref, arg_expr);
+			}
+		} else {
+			let mut new_arg_list: Vec<Expr> = Vec::new();
+			let mut new_arg_list_types: Vec<TypeRef> = Vec::new();
+			for i in 0..index_last_item {
+				new_arg_list.push(arg_list.pos_args.get(i).unwrap().clone());
+				new_arg_list_types.push(*arg_list_types.pos_args.get(i).unwrap());
+			}
```

Comment:
I refactored some of the code in this area to avoid the need for Clone and PartialEq implementations. Let me know if that's ok / if there's any issues with it

Reply:
It worked perfectly, thank you @Chriscbr